src/mainboard/clevo/mtl-h/variants/dgpu/gpio.c

```diff
@@ -310,20 +318,69 @@ static const struct pad_config gpio_table[] = {
 /* Pad configuration was generated automatically using intelp2m utility */
 void mainboard_configure_gpios(void)
 {
-	bool result;
-
-	gpio_configure_pads(gpio_table, ARRAY_SIZE(gpio_table));
-
-	/* dGPU power on sequence */
-	mdelay(4);
-	gpio_set(DGPU_PWR_EN, 1);
-	result = wait_ms(200, gpio_get(DGPU_PWRGD) == 1);
-	if (result) {
-		printk(BIOS_INFO, "dGPU powered on\n");
-		gpio_set(DGPU_RST_N, 1);
-	} else {
-		printk(BIOS_ERR, "dGPU failed to power on, turning off\n");
-		gpio_set(DGPU_PWR_EN, 0);
-	}
-	mdelay(50);
+    bool result;
+
+    gpio_configure_pads(gpio_table, ARRAY_SIZE(gpio_table));
+
+    if (dasharo_is_dgpu_enabled()) {
+        /* Ensure dGPU is fully powered off first */
+        gpio_set(DGPU_PWR_EN, 0);
+        mdelay(100);  // Ensure power rails fully discharge
+
+        /* Keep GPU in reset */
+        gpio_set(DGPU_RST_N, 0);
+        mdelay(10);  // Required delay before power-on
+
+        /* Enable PCIe SRCCLK */
+        p2sb_acpi_set_pci_clock(GPU_SRCCLK_PIN, true);
```

Comment:
this just generates ACPI bytecode, it doesn't actually directly control the srcclk